pyrival/graphs/maximum_matching.py

```diff
@@ -0,0 +1,65 @@
+import itertools
+from random import randint
+
+
+_DEFAULT_PRIME = 1073750017
+
+
+def maximum_matching(edges, mod=_DEFAULT_PRIME):
+    """
+    Returns the maximum cardinality matching of an undirected general graph
+    Uses a randomized algorithm to compute the rank of the Tutte matrix
+    The rank of the Tutte matrix is equal to twice the size of the maximum matching with high probability
+
+    Complexity: O(n ^ 3) worst case, O(n * m) with high constant factor on average
```

Comment:
Isn't it practically O(n^3) guaranteed? A better formula would be (maximum matching size) * n^2.

Reply:
Yup that'd be a better bound.